### DISCUSSION TITLE
#14 📑 Weekly Report 14

### DISCUSSION THREAD

**Discussion by fehlend0** [SE-TINF22B6/Statify#93] [Blog]
Hello and welcome to our weekly report!👋 We are excited to share the progress we’ve made this week.
## Cohesion as a software development metric:
First of all, we studied different metrics in software development this week. The one that our team presented was [cohesion](https://github.com/SE-TINF22B6/Statify/blob/main/Documentation/Cohesion_Statify.pdf). Here is a short summary of what we’ve learned about it:

Cohesion refers to the degree to which elements within a module work together to fulfill a single, well-defined purpose.
Cohesion should always be high, which means that a module/package/component is focused on only on what it should be doing, for example only methods relating to the intention of the class.

But how can you measure cohesion? We’ve discovered these two methods:

The first on is **LCOM4** (Lack of Cohesion in methods). It counts the number of disjoint sets of methods that do not share any instance variables. A lower LCOM4 value indicates higher cohesion, meaning methods within the class work together more closely.

The second method is **CBO** (Coupling Between Objects). It measures the degree of coupling between classes in a software system, indicating how many other classes a particular class relies on. A higher CBO value indicates higher coupling, meaning the class is more dependent on other classes.

And fortunately, there are such tools as **CodeScene** and **NDepend**, which allow you to do the calculation automatically and to generate reports. 

### Measuring the quality of our web application with Light House:
First we want to be sure, that our web application has a good quality. For this purpose we plan on using a metrics tool called Lighthouse. It is a Google tool that automates audits for improving web page quality in areas like performance, accessibility, SEO, and best practices. It provides metrics such as page load time and checks for common issues like color contrast and proper labeling. Available as a Chrome extension, command line tool, and Node module, Lighthouse generates detailed reports with actionable insights to enhance web applications.

### Test Coverage 
Additionally, we want to measure the test coverage in our project. It helps to identify untested parts of the code, ensuring comprehensive testing and reducing bugs. It improves code quality by catching errors early and supports safe refactoring by providing a safety net. High coverage also boosts stakeholder confidence and supports continuous integration by ensuring new changes don’t break existing functionality. Lastly, it helps meet regulatory requirements in certain industries.

### Execution Quote of Test Cases
Finally, we want to measure the quote of execution of test cases. It is crucial for ensuring the overall quality and reliability of the software. It helps track testing progress, providing insights into which areas need more focus and resources. This measurement also aids in risk management by identifying frequent failures and potential issues early. Additionally, it supports continuous improvement by allowing teams to analyze and refine their testing processes based on concrete data.

## Development

It looks like we’ve achieved a great milestone in our project – we are done with the frontend implementations!🎉
Our last steps there were the implementation of the track page and of the component for statistics generation. The results are shown below:

![trackPage](https://github.com/SE-TINF22B6/Statify/assets/106541016/3e65c994-472e-4182-bfbf-78bb11f5e249)
![generate](https://github.com/SE-TINF22B6/Statify/assets/106541016/d71f3c37-820c-480e-a50b-880c1dda7b4b)

Let us continue this journey and look where we will be next week,
Your Spotify Team

**Comment by mirak55** [comment]
Dear Statify team,
nice work on the topic of cohesion, you gave an good overview. The metrics you chose for your project were also well thought through so also nice work on this and on the explanations why you chose them and how you want to implement the metrics. Also awesome work on finishing your frontend, it looks really good and modern. 
Keep up the good work!

**Comment by MaarcS** [comment]
Dear Statify team,

your description of cohesion is very informative and well explained. Also thank you for recommending NDepend. We have used it to analyze the code quality of our backend code, which worked fine and gave us a very detailed overview.

It is nice to see how your project is turning out. Your user interface looks very user friendly and well thought out.

Your CookHub Team
